NAA_Analysis.py

```diff
@@ -58,7 +58,51 @@ def peak_finder(measurement):
     return peaks_found
 
 
-def count_rate(M, B, energy):
+def NAA_net_area(measurement, energy):
+    """
+    NAA_net_area determines the net area of a given peak without the use
+    of compton regions. It generates a region of interest (ROI) based on
+    full width half maximum (FWHM). A line is generated using the edges of the
+    ROI and an integral of the line is determined. The gross counts of the peak
+    are then subtracted by the integral and a net area and uncertainty are
+    determined and returned.
+    """
+    E0 = measurement.energy_cal[0]
+    Eslope = measurement.energy_cal[1]
+    sample_counts = measurement.data
+    energy_channel = int((energy - E0) / Eslope)
+
+    region_size = 1.3
+    # Rough estimate of FWHM.
+    fwhm = 0.05*energy**0.5
+    fwhm_channel = int(region_size * (fwhm - E0) / Eslope)
+    # peak gross area
+    gross_counts_peak = sum(sample_counts[(energy_channel - fwhm_channel):
+                                          (energy_channel + fwhm_channel)])
+    peak_channels = measurement.channel[(energy_channel - fwhm_channel):
+                                        (energy_channel + fwhm_channel)]
+    # first and last channel counts of peak
+    start_peak_c = sample_counts[(energy_channel - fwhm_channel)]
+    end_peak_c = sample_counts[(energy_channel + fwhm_channel)]
+    first_last_channel = [(energy_channel - fwhm_channel),
+                          (energy_channel + fwhm_channel)]
+
+    # generate line under peak
+    a_matrix = np.vstack([first_last_channel, np.ones(2)]).T
+    cslope, c0 = np.linalg.lstsq(a_matrix, [start_peak_c, end_peak_c])[0]
+
+    # integrate line under peak and acquire net area
```

Comment:
A line integral is a nice way to visualize the Compton subtraction, but the result is identical to just taking the average of the first and last bin, and letting that be the Compton subtraction for each bin. So you can just do:
```
compton_area = (start_peak_c + end_peak_c) / 2 * len(peak_channels)
net_area = gross_counts_peak - compton_area
```